nav2_dwb_controller/nav_2d_utils/include/nav_2d_utils/odom_subscriber.hpp

```diff
@@ -58,14 +60,13 @@ class OdomSubscriber
    * @param nh NodeHandle for creating subscriber
    * @param default_topic Name of the topic that will be loaded of the odom_topic param is not set.
    */
-  explicit OdomSubscriber(rclcpp::Node & nh, std::string default_topic = "odom")
+  explicit OdomSubscriber(nav2_lifecycle::LifecycleNode & nh, std::string default_topic = "odom")
   {
     std::string odom_topic;
     nh.get_parameter_or("odom_topic", odom_topic, default_topic);
     odom_sub_ =
       nh.create_subscription<nav_msgs::msg::Odometry>(odom_topic,
-        [&](const nav_msgs::msg::Odometry::SharedPtr msg) {odomCallback(msg);},
-        1);
+        std::bind(&OdomSubscriber::odomCallback, this, std::placeholders::_1));
```

Comment:
I prefer the lambda syntax and it is supposedly more efficient as well.

Reply:
This usage - creating a lambda function to wrap an already existing class method - didn't make much sense to me. It's just introducing an unnecessary level of indirection (less efficient). If we've already got a class method, we should simply use that directly. Otherwise, a lambda would be find if it had the contents of odomCallback inlined, but that's not the case here.